src/indexes.ts

```diff
@@ -262,13 +256,13 @@ class Index<T extends Record<string, any> = Record<string, any>> {
    * @param data - Data to update
    * @returns Promise to the current Index object with updated information
    */
-  async update(data: IndexOptions): Promise<EnqueuedTask> {
-    const url = `indexes/${this.uid}`;
-    const task = await this.httpRequest.patch(url, data);
-
-    task.enqueuedAt = new Date(task.enqueuedAt);
+  async update(data?: IndexOptions): Promise<EnqueuedTask> {
```

Comment:
Is there a reason to allow updating without any indexOptions?

Reply:
Oh, `IndexOptions` has one optional parameter `primaryKey?: string`. So that's probably why I did this. But instead I'm just going to make this property non-optional.

Reply:
Actually that would break something else. I'm just reverting it for now to required parameter.